### PR TITLE
RIA-6780: Fix issue with parsing 'hearing date'

### PR DESCRIPTION
**Pull request by augustio**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-6780

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/detentionengagementteam/DetentionEngagementTeamRequestResponseReviewPersonalisation.java [hunk 1]
```diff
@@ -71,9 +73,8 @@ public String getReferenceId(Long caseId) {
     public Map<String, Object> getPersonalisationForLink(AsylumCase asylumCase) {
         requireNonNull(asylumCase, "asylumCase must not be null");
 
-        String hearingDate = LocalDate.parse(asylumCase.read(LIST_CASE_HEARING_DATE, String.class)
-                .orElseThrow(() -> new IllegalStateException("Hearing date is not present")))
-            .format(DateTimeFormatter.ofPattern("d MMM yyyy"));
+        String hearingDate = dateTimeExtractor.extractHearingDate(asylumCase.read(LIST_CASE_HEARING_DATE, String.class)
+            .orElseThrow(() -> new IllegalStateException("Hearing date is not present")));
```
Comment (hussain-faruk) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/detentionengagementteam/DetentionEngagementTeamRequestResponseReviewPersonalisation.java:77]: Shouldnt this be
dateTimeExtractor.extractHearingDate(hearingDetailsFinder.getHearingDateTime(asylumCase))

I believe the hearingDetailsFinder.getHearingDateTime method already throws the exception too.

Comment (augustio) [reply src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/detentionengagementteam/DetentionEngagementTeamRequestResponseReviewPersonalisation.java:77]: I didn't use the HearingDetailsFinder::getHaringDateTime. Maybe I should use that.

Comment (augustio) [reply src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/detentionengagementteam/DetentionEngagementTeamRequestResponseReviewPersonalisation.java]: I have taken care of this by using  HearingDetailsFinder::getHaringDateTime to get the hearing date. Thanks.

**Review by hussain-faruk — DISMISSED** [review]
LGTM

**Review by hussain-faruk — APPROVED** [review]

**Review by Dbarnes-v1 — APPROVED** [review]

**Pull request merged**